aws-lc-rs/src/rsa.rs

```diff
@@ -507,69 +546,99 @@ impl RsaParameters {
 
 #[inline]
 #[allow(non_snake_case)]
-unsafe fn build_public_RSA(public_key: &[u8]) -> Result<LcPtr<*mut RSA>, Unspecified> {
+unsafe fn build_public_RSA(public_key: &[u8]) -> Result<LcPtr<*mut EVP_PKEY>, Unspecified> {
     let mut cbs = cbs::build_CBS(public_key);
 
-    let rsa = LcPtr::new(RSA_parse_public_key(&mut cbs))?;
-    Ok(rsa)
+    let rsa = DetachableLcPtr::new(RSA_parse_public_key(&mut cbs))?;
+
+    let pkey = LcPtr::new(EVP_PKEY_new())?;
+
+    if 1 != EVP_PKEY_assign_RSA(*pkey, *rsa) {
+        return Err(Unspecified);
+    }
+
+    rsa.detach();
+
+    Ok(pkey)
 }
 
 #[inline]
 #[allow(non_snake_case)]
-unsafe fn build_private_RSA(public_key: &[u8]) -> Result<LcPtr<*mut RSA>, KeyRejected> {
-    let mut cbs = cbs::build_CBS(public_key);
+unsafe fn build_private_RSA(private_key: &[u8]) -> Result<LcPtr<*mut EVP_PKEY>, KeyRejected> {
```

Comment:
Should we update the function naming for both public and private? something like `build_private_RSA_PKEY`